### DISCUSSION TITLE
Quickly backup toolbar callouts to save in a note

### DISCUSSION THREAD

**Discussion by calvin-1** [chrisgurney/obsidian-note-toolbar#185] [Scripting]
First of all, thank you for your work on this plugin!

I'm just going through a situation where I made changes to my toolbars on one device but it was overwritten by the old version after syncing. Not sure why, but I was able to recover the toolbars by copying them as callouts from my phone.

My suggestion is to add a quick way to copy all toolbars as callouts to paste in a md note as a backup.. A step further would be to save all toolbar callouts to a markdown file automatically.. I'm not sure if it's possible for a plugin to write to a note but anyway.

I did a quick search and checked the roadmap - sorry if this has been suggested before! I know my situation is unique but a backup feature would be great!

**Comment by chrisgurney** [comment]
@calvin-1 Hi there, thank you for the suggestion! 

I think that's a good idea. 

I'm not fully trusting of syncing myself, and I use Obsidian Sync. 

Even if just the toolbars are backed up (without the mappings) that would hopefully save the majority of the effort of reconstructing the plugin's configuration.

**Comment by chrisgurney** [comment]
@calvin-1 Sorry this took so long, but now that Note Toolbar has an API, I made it a priority to support your idea with the first couple of toolbar methods. To use it:

1.) Update to the latest version
2.) You'll need Dataview or JS Engine (though this could be modified for Templater as well)
3.) Copy this script into your vault and read the instructions:
https://github.com/chrisgurney/obsidian-note-toolbar/blob/master/examples/Scripts/Dataview/NtbExport.js

Let me know if you have any questions!
